crates/sui-indexer-alt-framework/src/pipeline/concurrent/commit_watermark.rs

```diff
@@ -255,3 +255,303 @@ pub(super) fn commit_watermark<H: Handler + 'static>(
         );
     })
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{pipeline::Processor, MIGRATIONS};
+    use prometheus::Registry;
+    use std::time::Duration;
+    use sui_field_count::FieldCount;
+    use sui_pg_db::{self as db, temp::TempDb, Db, DbArgs};
+    use tokio::sync::mpsc;
+
+    struct TestHandler;
+
+    struct Entry;
+
+    impl FieldCount for Entry {
+        const FIELD_COUNT: usize = 1;
+    }
```

Comment:
Note that I've put out #20564 which would allow you to use the derive macro on this type, maybe? Not sure if the macro would complain about there not being a field as well.